jdaviz/configs/cubeviz/plugins/sonify_data/sonify_data.py

```diff
@@ -0,0 +1,118 @@
+
+
+@tray_registry('cubeviz-sonify-data', label="Sonify Data",
+               viewer_requirements=['spectrum', 'image'])
+class SonifyData(PluginTemplateMixin, DatasetSelectMixin, SpectralSubsetSelectMixin):
+    """
+    See the :ref:`Sonify Data <cubeviz-sonify-data>` for more details.
+
+    Only the following attributes and methods are available through the
+    :ref:`public plugin API <plugin-apis>`:
+
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.show`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.open_in_tray`
+    * :meth:`~jdaviz.core.template_mixin.PluginTemplateMixin.close_in_tray`
+    """
+    template_file = __file__, "sonify_data.vue"
+
+    sample_rate = IntHandleEmpty(44100).tag(sync=True)
+    buffer_size = IntHandleEmpty(2048).tag(sync=True)
+    assidx = FloatHandleEmpty(2.5).tag(sync=True)
+    ssvidx = FloatHandleEmpty(0.65).tag(sync=True)
+    eln = Bool(False).tag(sync=True)
+    audfrqmin = FloatHandleEmpty(50).tag(sync=True)
+    audfrqmax = FloatHandleEmpty(1500).tag(sync=True)
+    pccut = IntHandleEmpty(20).tag(sync=True)
+    volume = IntHandleEmpty(100).tag(sync=True)
+    stream_active = Bool(True).tag(sync=True)
+    has_strauss = Bool(_has_strauss).tag(sync=True)
+
+    # TODO: can we referesh the list, so sounddevices are up-to-date when dropdown clicked?
+    sound_devices_items = List().tag(sync=True)
+    sound_devices_selected = Unicode('').tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._plugin_description = 'Sonify a data cube'
+        self.docs_description = 'Sonify a data cube using the Strauss package.'
+
+        if self.has_strauss:
+            devices, indexes = self.build_device_lists()
+            self.sound_device_indexes = dict(zip(devices, indexes))
+            self.sound_devices_items = devices
+            self.sound_devices_selected = dict(zip(indexes, devices))[sd.default.device[1]]
+
+        # TODO: Remove hardcoded range and flux viewer
+        self.spec_viewer = self.app.get_viewer('spectrum-viewer')
+        self.flux_viewer = self.app.get_viewer('flux-viewer')
+
+    @property
+    def user_api(self):
+        expose = []
+        return PluginUserApi(self, expose)
+
+    @with_spinner()
+    def vue_sonify_cube(self, *args):
+        # Get index of selected device
+        selected_device_index = self.sound_device_indexes[self.sound_devices_selected]
+        self.flux_viewer.get_sonified_cube(self.sample_rate, self.buffer_size,
+                                           selected_device_index, self.assidx, self.ssvidx,
+                                           self.pccut, self.audfrqmin,
+                                           self.audfrqmax, self.eln)
+
+        # Automatically select spectrum-at-spaxel tool
+        spec_at_spaxel_tool = self.flux_viewer.toolbar.tools['jdaviz:spectrumperspaxel']
+        self.flux_viewer.toolbar.active_tool = spec_at_spaxel_tool
```

Comment:
I can't decide if this is convenient or could be confusing, it isn't a pattern we currently use anywhere although I think we have discussed the ability to control tool selection from plugins.  Maybe @Jenneh will have thoughts (whether the "spectrum at spaxel" tool in the flux cube viewer should automatically activate after sonification is complete, perhaps deactivating any other tool the user had enabled previously).

Reply:
I agree that it is not expected behavior but early user testing informed us that we needed to automatically have the sound play after the user presses sonify data and the audified cube is loaded. I can create a follow-up ticket to decide exactly how we want to do this.

Reply:
ok, we can do this for now, but let's please revisit.  Maybe (eventually) a message with a button in the plugin itself to first activate the tool would be ideal to help teach how to toggle it later - we had considered this for other plugins as well but don't yet have the infrastructure to do that.

Reply:
This may become a moot point once we have the sonified cube as its own layer in the viewer. We will still need to instruct the user how to get the sound to turn on/off (depending on the default behavior) but by that point it will be out of the spectrum-at-spaxel tool. Related tickets #3329 #3330 #3331